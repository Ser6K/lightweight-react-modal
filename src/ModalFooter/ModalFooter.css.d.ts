declare namespace ModalFooterCssNamespace {
  export interface IModalFooterCss {
    footer: string;
  }
}

declare const ModalFooterCssModule: ModalFooterCssNamespace.IModalFooterCss & {
  /** WARNING: Only available when `css-loader` is used without `style-loader` or `mini-css-extract-plugin` */
  locals: ModalFooterCssNamespace.IModalFooterCss;
};

export = ModalFooterCssModule;
