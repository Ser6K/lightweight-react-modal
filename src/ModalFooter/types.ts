import { ClassNameType, ReactNodeType } from 'src/types'

export interface ModalFooterTypes {
  children: ReactNodeType
  className?: ClassNameType
  [propName: string]: any
}
