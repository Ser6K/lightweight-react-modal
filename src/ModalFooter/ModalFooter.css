.footer {
    flex-shrink: 0;
    border-top: 1px solid grey;
    padding: 15px 20px;
    box-sizing: border-box;
}
