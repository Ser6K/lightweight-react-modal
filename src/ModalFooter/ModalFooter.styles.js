export default {
    footer: {
        flexShrink: 0,
        borderTop: '1px solid grey',
        padding: '15px 20px',
        boxSizing: 'borderBox',
    },
};
