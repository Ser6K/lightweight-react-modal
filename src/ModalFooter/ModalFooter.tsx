import React from 'react'
import { classnames } from 'src/utils'

import { ModalFooterTypes } from './types'

import styles from './ModalFooter.css'

const ModalFooter: React.FC<ModalFooterTypes> = ({ children, className, ...props }) => (
  <div {...props} className={classnames(styles.footer, className)}>
    {children}
  </div>
)

export default ModalFooter
