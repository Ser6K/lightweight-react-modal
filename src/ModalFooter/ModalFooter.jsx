import React from 'react';
import PropTypes from 'prop-types';
import { FooterDisplayName } from 'utils';
import { Footer } from './ModalFooter.styles';

const ModalFooter = React.memo(({ children, ...props }) => (
    <Footer {...props}>
        {children}
    </Footer>
));

ModalFooter.propTypes = {
    children: PropTypes.node.isRequired,
};

ModalFooter.displayName = FooterDisplayName;
export default ModalFooter;
