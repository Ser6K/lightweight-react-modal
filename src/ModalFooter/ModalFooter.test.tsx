import React from 'react'
import { mount } from 'enzyme'
import ModalFooter from './ModalFooter'

describe('Modal footer', () => {
  test('it matches snapshot with no props', () => {
    const modalFooter = mount(<ModalFooter>Footer</ModalFooter>)
    expect(modalFooter).toMatchSnapshot()
  })
  test('it matches snapshot with props', () => {
    const modalFooter = mount(
      <ModalFooter className="className" style={{ display: 'inline' }}>
        Footer
      </ModalFooter>
    )
    expect(modalFooter).toMatchSnapshot()
  })
})
