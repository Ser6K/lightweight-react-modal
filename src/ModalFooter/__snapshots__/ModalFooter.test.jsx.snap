// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal footer it matches snapshot with no props 1`] = `
<ModalFooter>
  <ModalFooterstyles__Footer>
    <div
      className="ModalFooterstyles__Footer-sc-1ls2gha-0 gkRbig"
    >
      Footer
    </div>
  </ModalFooterstyles__Footer>
</ModalFooter>
`;

exports[`Modal footer it matches snapshot with props 1`] = `
<ModalFooter
  className="className"
  style={
    Object {
      "display": "inline",
    }
  }
>
  <ModalFooterstyles__Footer
    className="className"
    style={
      Object {
        "display": "inline",
      }
    }
  >
    <div
      className="ModalFooterstyles__Footer-sc-1ls2gha-0 gkRbig className"
      style={
        Object {
          "display": "inline",
        }
      }
    >
      Footer
    </div>
  </ModalFooterstyles__Footer>
</ModalFooter>
`;
